indra_db/reading/read_db.py

```diff
@@ -181,54 +176,34 @@ def get_cols():
         return (batch_id, self.reading_id, self.db_info_id,
-                self.statement.uuid, self.statement.get_hash(),
-                self.statement.evidence[0].get_source_hash(),
-                self.statement.__class__.__name__,
-                json.dumps(self.statement.to_json()), self.indra_version,
+                self.result.uuid, self.result.get_hash(),
+                self.result.evidence[0].get_source_hash(),
+                self.result.__class__.__name__,
+                json.dumps(self.result.to_json()), self.indra_version,
                 self._get_text_hash())
 
-
-def get_stmts_safely(reading_data):
-    stmt_data_list = []
-    try:
-        stmts = reading_data.get_statements()
-    except Exception as e:
-        logger.error("Got exception creating statements for %d."
-                     % reading_data.reading_id)
-        logger.exception(e)
-        return []
-    if stmts is not None:
-        if not len(stmts):
-            logger.debug("Got no statements for %s." % reading_data.reading_id)
-        for stmt in stmts:
-            stmt.evidence[0].pmid = None
-            stmt_data = DatabaseStatementData(stmt, reading_data.reading_id)
-            stmt_data_list.append(stmt_data)
-    else:
-        logger.warning("Got None statements for %s." % reading_data.reading_id)
-    return stmt_data_list
+    def _get_text_hash(self):
+        ev = self.result.evidence[0]
+        simple_text = self.__text_patt.sub('', ev.text)
+        if 'coords' in ev.annotations.keys():
+            simple_text += str(ev.annotations['coords'])
+        return make_hash(simple_text.lower(), 16)
 
 
-def make_statements(reading_data_list, num_proc=1):
-    """Convert a list of ReadingData instances into StatementData instances."""
-    stmt_data_list = []
+class DatabaseMeshRefData(DatabaseResultData):
+    def __init__(self, result, reading_id=None, db_info_id=None,
+                 indra_version=None):
+        super().__init__(result, reading_id, db_info_id, indra_version)
+        self.pmid, self.mesh_id = result
 
-    if num_proc is 1:  # Don't use pool if not needed.
-        for reading_data in reading_data_list:
-            stmt_data_list += get_stmts_safely(reading_data)
-    else:
-        pool = Pool(num_proc)
-        try:
-            stmt_data_list_list = pool.map(get_stmts_safely, reading_data_list)
-            for stmt_data_sublist in stmt_data_list_list:
-                stmt_data_list += stmt_data_sublist
-        finally:
-            pool.close()
-            pool.join()
+    @staticmethod
+    def get_cols():
+        """Get the columns for the tuple returned by `make_tuple`."""
+        return 'pmid_num', 'mesh_num'
 
-    logger.info("Found %d statements from %d readings." %
-                (len(stmt_data_list), len(reading_data_list)))
-    return stmt_data_list
+    def make_tuple(self, batch_id):
+        """Make a tuple for copying into the database."""
+        return tuple(self.result)
```

Comment:
A couple of tweaks here. First, `self.results` is no longer a parameter so the tuple will need to be spelled out. Also, unless I am mistaken, the mesh_id is a string at this point, so it will need to be int-ified:
```
int(self.mesh_id[1:])
```

Reply:
The mesh_id is being int-ified while being parsed by MTIReader, so by arriving here it is not a string anymore